### DISCUSSION TITLE
api key authentication in browser?

### DISCUSSION THREAD

**Discussion by ceceayo** [deta/cloud-docs#432] [Help]
how do I log in with my api key from chrome into my python micro that is using flask?

**Comment by abdelhai** [comment]
this is not possible. you need to use/write an authentication service.

**Comment by ceceayo** [comment]
would it be easier if i'd were to use flask basic authentication?

**Reply by Sokhavuth** [reply]
You could use PyJWT to create a JSON web token and save it as a cookie or store it in the database as a session for authentication.
[https://pyjwt.readthedocs.io/en/stable/](https://pyjwt.readthedocs.io/en/stable/)

**Comment by Wallvon** [comment]
I'd suggest just using [Deta Base](https://docs.deta.sh/docs/base/about) and storing something like a token or a string as a cookie and in a base. For the former, you could use an encoded JWT, for example, and for the latter, you could just store the string as a key and any further information you need about a user as the value. Do keep in mind that the latter will force you to make requests to the base every time a user goes to a page, whereas with the JWT, you just store basic user information in there.
